Clarify install of Gateway API for container runner SSH reruns [ONPREM-608] (#8485)
circleci/circleci-docs: jekyll/_cci2/container-runner-installation.adoc

NOTE: **Supported Gateway API implementations:**
CircleCI has tested, and currently supports link:https://gateway.envoyproxy.io/[Envoy Gateway] as an implementation for the link:https://gateway-api.sigs.k8s.io/[Gateway API]. Other Gateway API implementations that support TCPRoute resources are likely compatible, but not all of these implementations have not been tested, and may not be compatible. Our recommendation is to use link:https://gateway.envoyproxy.io/[Envoy Gateway] as your implementation.

==== 1. Install Envoy Gateway to your cluster
. First, install the Gateway API CRDs and Envoy Gateway, as defined in the
link:https://gateway.envoyproxy.io/latest/install/install-yaml/[Envoy Gateway installation document]:
+
[source,shell]
----
kubectl apply -f https://github.com/envoyproxy/gateway/releases/download/latest/install.yaml
----

==== 2. Enable the Rerun job with SSH feature

> . Once all Gateway API prerequisites are installed and available, add the `agent.ssh.enabled = “true”` parameter to the container runner configuration:

Reviewer: Ah we should fix these magic quotes while we can

```suggestion
. Once all Gateway API prerequisites are installed and available, add the `agent.ssh.enabled = "true"` parameter to the container runner configuration:
```

NOTE: **Supported Gateway API implementations:**

> CircleCI has tested, and currently supports link:https://gateway.envoyproxy.io/[Envoy Gateway] as an implementation for the link:https://gateway-api.sigs.k8s.io/[Gateway API]. Other link:https://gateway-api.sigs.k8s.io/implementations/[Gateway API implementations] that support TCPRoute resources are likely compatible, but not all of these implementations have been tested, and may not be compatible. Our recommendation is to use Envoy Gateway as your implementation. As with most Gateway API implementations, note that Envoy Gateway is currently in beta and is still under development.

Reviewer 2: ```suggestion
CircleCI has tested, and currently supports link:https://gateway.envoyproxy.io/[Envoy Gateway] as an implementation for the link:https://gateway-api.sigs.k8s.io/[Gateway API]. Other link:https://gateway-api.sigs.k8s.io/implementations/[Gateway API implementations] that support TCPRoute resources are likely to be compatible, but not all of these implementations have been tested, and therefore compatibility is not guaranteed. Our recommendation is to use Envoy Gateway as your implementation. Envoy Gateway is currently in beta and is under development.
```
